### DISCUSSION TITLE
[Help needed persist] Type error: Expression produces a union type that is too complex to represent

### DISCUSSION THREAD

**Discussion by AndreiBarditaWolf** [pmndrs/zustand#1249] [Q&A]
EDIT: Fixed the issue, the problem was that my typescript version was a little older (4.4.4) so i upgraded to the latest stable version of ts (4.7) with `npm i typescript@4.7`

Hello. I'm using zustand with a nextjs application for changes between languages. Everything works fine, no issues whatsoever on the local development environment with the persist middleware

Problem comes up when i build the app, i get the error: ```Type error: Expression produces a union type that is too complex to represent.```

The language.ts file:

```
import create from "zustand";
import { persist } from "zustand/middleware";

interface State {
  language: string;
  setLanguage: (lang: string) => void;
}

const useLanguageStore = create<State>()(
  persist(
    (set) => ({
      language: "en",
      setLanguage: (lang) => set((state) => ({ language: lang })),
    }),
    {
      name: "storage",
    }
  )
);

export default useLanguageStore;
```

HINT1! : I dont know why but with the exact and really exact same configuration, interface and all, on a new plain nextjs project i've started it works really well, local and built.

HINT2! : If i change the file to language.js (Javascript, not Typescript), the build works and it gets compiled. I use it as a temporary solution, but i would be really happy if someone could help me figure it out.

Below is a screenshot with the console error. The same error repeats about 3 to 4 times and then stops the console, showing the second image below:
![unknown](https://user-images.githubusercontent.com/102662067/187952730-cfe8478c-07ef-45f8-865d-880894ce8ef5.png)
![Screenshot 2022-09-01 182949](https://user-images.githubusercontent.com/102662067/187953340-a26ff925-9acf-49ee-9aca-12f7572c276a.png)

**Comment by dai-shi** [comment]
Maybe your TS version is a little old? https://github.com/pmndrs/zustand/issues/1013

**Reply by AndreiBarditaWolf** [reply]
do you know how can i upgrade nextjs's typescript version? cant really find anything that could help

**Reply by dai-shi** [reply]
`npm install typescript`? I'm not very familiar with nextjs.
Which version of TS are you currently using?

**Reply by AndreiBarditaWolf** [reply]
> `npm install typescript`? I'm not very familiar with nextjs. Which version of TS are you currently using?

version 4.4.4

just upgraded to 4.7 and works like a charm, thank you. didnt think about this

**Reply by dai-shi** [reply]
Cool.

https://github.com/pmndrs/zustand/issues/1013#issuecomment-1235461649 It seems like v4.5.5 works at least.
